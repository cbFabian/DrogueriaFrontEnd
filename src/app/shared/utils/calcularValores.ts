export const calcularValores = (producto: string, 
    reunion: string, esFestivo: boolean) => {
        if(!producto && !reunion){
            return 0;
        }
        if(!producto){
            return JSON.parse(reunion).precio;
        }
        if(!reunion){
            return JSON.parse(producto).precio;
        }
        let valorTotal: number = JSON.parse(producto).precio + JSON.parse(reunion).precio;
        if(esFestivo){
            valorTotal *= 2;
        }                
        return valorTotal;
};
