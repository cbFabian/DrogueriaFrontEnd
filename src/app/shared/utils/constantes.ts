export const constantes = {
    HORAS24: 24,
    HORA15: 15,
    HORA19: 19,
    MINUTOS60: 60,
    SEGUNDOS60: 60,
    MILISEGUNDOS: 1000,
    POSICION_DOS: 2,
    POSICION_TRES: 3,
    POSICION_CUATRO: 4,
    NUMERO_DOS: 2,
    NUMERO_TRES: 3,
    HORAMINIMA: 4,
    NUMERO_CINCO: 5,
    HORAMAXIMA: 8,
};
