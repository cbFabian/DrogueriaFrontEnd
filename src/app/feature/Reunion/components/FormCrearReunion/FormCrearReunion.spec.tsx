import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { FormCrearReunion } from './';
import { setTextEvent } from 'app/shared/utils/test';

describe('FormCrearReunion test', () => {
    let componentWrapper: RenderResult;
    let componentProps: React.ComponentProps<typeof FormCrearReunion> & {
        onSubmit: SinonStub;
    };

    beforeEach(() => {
        componentProps = {
          formTitle: 'Form test',
          onSubmit: stub(),
        };
        componentWrapper = render(<FormCrearReunion {...componentProps} />);
    });

    it('should match snapshot', () => {
        expect(componentWrapper.container).toMatchSnapshot();
    });

    it('should fail on submit all fields missing', async () => {
        const elem = componentWrapper.container;
        const submitButton = elem.querySelector('button[type="submit"]');
        const DOS = 2;

        await wait(() => {
          submitButton && fireEvent.click(submitButton);
        });
        const spans = elem.querySelectorAll('span');
        expect(spans.length).toBe(DOS);
        expect(spans[0].textContent).toBe('El campo tipo es requerido.');
        expect(spans[1].textContent).toBe('El campo precio es requerido.');
      });
    
      it('should fail on submit one field missing', async () => {
        const elem = componentWrapper.container;
    
        const tipo = elem.querySelector('input[name="tipo"]');
        const submitButton = elem.querySelector('button[type="submit"]');
    
        await wait(() => {
          tipo && fireEvent.change(tipo, setTextEvent('tipo', 'Lorem'));
        });
    
        await wait(() => {
          submitButton && fireEvent.click(submitButton);
        });
        const spans = elem.querySelectorAll('span');
        expect(spans.length).toBe(1);
        expect(spans[0].textContent).toBe('El campo precio es requerido.');
      });
    
      it('should submit', async () => {
        const elem = componentWrapper.container;
        const VEINTI4MIL = 24000;

        const tipo = elem.querySelector('input[name="tipo"]');
        const precio = elem.querySelector('input[name="precio"]');
        const submitButton = elem.querySelector('button[type="submit"]');
    
        await wait(() => {
            tipo && fireEvent.change(tipo, setTextEvent('tipo', 'Lorem'));
        });
        await wait(() => {
            precio && fireEvent.change(precio, setTextEvent('precio', '24000'));
        });
        await wait(() => {
          submitButton && fireEvent.click(submitButton);
        });
    
        const formSubmitted = componentProps.onSubmit.firstCall.args[0];
    
        expect(formSubmitted.tipo).toBe('Lorem');
        expect(formSubmitted.precio).toBe(VEINTI4MIL);
      });
});
