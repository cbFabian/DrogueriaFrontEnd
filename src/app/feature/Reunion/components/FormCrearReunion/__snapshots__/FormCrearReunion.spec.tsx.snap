// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCrearReunion test should match snapshot 1`] = `
<div>
  <form>
    <h2>
      Form test
    </h2>
    <input
      class="sc-AxirZ eAukyj"
      name="tipo"
      placeholder="Tipo"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="precio"
      placeholder="Precio"
      value=""
    />
    <button
      class="sc-AxjAm kgxTKy"
      type="submit"
    >
      Registrar
    </button>
  </form>
</div>
`;
